WalletWasabi/Crypto/ZeroKnowledge/ProofSystem.cs

```diff
@@ -40,5 +41,19 @@ public static Statement IssuerParameters(CoordinatorParameters iparams, MAC mac,
 				{ Generators.GV - iparams.I, O,             O,              Generators.Gx0, Generators.Gx1, Generators.Ga },
 				{ iparams.Cw,                Generators.Gw, Generators.Gwp, O,              O,              O },
 			});
+
+		public static Knowledge ShowCredential(CredentialPresentation c, Scalar z, Scalar t, Scalar a, Scalar r, CoordinatorParameters iparams)
+			=> new Knowledge(ShowCredential(c, z * iparams.I, iparams), new ScalarVector(z, (t * z).Negate(), t, a, r));
```

Comment:
```suggestion
		public static Knowledge ShowCredential(CredentialPresentation presentation, Credential credential, Scalar z, CoordinatorParameters iparams)
			=> new Knowledge(ShowCredential(presentation, z * iparams.I, iparams), new ScalarVector(z, (credential.Mac.T * z).Negate(), credential.Mac.T, credential.Amount, credential.Randomness));
```

I would go further than that and also change the argument order to put iparams in front of the secret inputs

Reply:
I had this change but forgot to push it.

Reply:
About the arguments order it is hard to decide because:

* _Semantic_: a method called `DoSomethingWithThing` should receive `Thing theThing` as first argument. In pseudocode this makes sense `ThrowRock(Rock theRock, Angle angle, Velocity initialVelocity)` but this makes no so much sense `ThrowRock(Angle angle, Rock theRock, Velocity initialVelocity)`.

* _Consistency_: (or Usability) if we put it first here then we have to put it first in other places too and it could make no sense in same places.

Reply:
let's revisit later, with fresh eyes we can do another cleanup pass after we've merged all the proof stuff